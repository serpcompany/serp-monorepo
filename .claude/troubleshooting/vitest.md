https://vitest.dev/guide/common-errors.html