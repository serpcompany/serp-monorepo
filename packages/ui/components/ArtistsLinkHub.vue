<template>
  <div
    class="mt-2 grid grid-cols-1 gap-2 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4"
  >
    <div v-for="artist in artists" :key="artist.slug">
      <span>
        <NuxtLink :to="`/artists/${artist.slug}/`" aria-label="artist name">
          {{ artist.name }}
        </NuxtLink>
      </span>
    </div>
  </div>
</template>

<script setup lang="ts">
import type { ArtistBase } from '@serp/types/types';

const props = defineProps({
  artists: {
    type: Array as () => ArtistBase[],
    required: true
  }
});
</script>
