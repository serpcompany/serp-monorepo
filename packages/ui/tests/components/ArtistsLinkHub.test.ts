import { describe, expect, it } from 'vitest';
import ArtistsLinkHub from '../../components/ArtistsLinkHub.vue';
import ComponentRender from '../componentRender';

describe('artistsLinkHub Snapshot', () => {
  const baseArtists = [
    { slug: 'artist-1', name: 'Artist 1' },
    { slug: 'artist-2', name: 'Artist 2' },
    { slug: 'artist-3', name: 'Artist 3' },
  ];

  it.each([
    ['with no artists', { props: { artists: [] } }],
    ['with one artist', { props: { artists: [baseArtists[0]] } }],
    ['with multiple artists', { props: { artists: baseArtists } }],
  ])(
    'renders %s correctly',
    async (desc: string, options: { props: unknown }) => {
      const html = await ComponentRender(
        `ArtistsLinkHub ${desc}`,
        options,
        ArtistsLinkHub,
      );
      expect(html).toMatchSnapshot();
    },
  );
});
