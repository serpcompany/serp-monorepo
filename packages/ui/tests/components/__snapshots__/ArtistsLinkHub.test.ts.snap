// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ArtistsLinkHub Snapshot > renders with multiple artists correctly 1`] = `
<div class="
  gap-2
  grid
  grid-cols-1
  lg:grid-cols-3
  md:grid-cols-2
  mt-2
  xl:grid-cols-4
">
  <div>
    <span>
      <a
        aria-label="artist name"
        href="/artists/artist-1/"
      >Artist 1</a>
    </span>
  </div>
  <div>
    <span>
      <a
        aria-label="artist name"
        href="/artists/artist-2/"
      >Artist 2</a>
    </span>
  </div>
  <div>
    <span>
      <a
        aria-label="artist name"
        href="/artists/artist-3/"
      >Artist 3</a>
    </span>
  </div>
</div>
`;

exports[`ArtistsLinkHub Snapshot > renders with no artists correctly 1`] = `
<div class="
  gap-2
  grid
  grid-cols-1
  lg:grid-cols-3
  md:grid-cols-2
  mt-2
  xl:grid-cols-4
"></div>
`;

exports[`ArtistsLinkHub Snapshot > renders with one artist correctly 1`] = `
<div class="
  gap-2
  grid
  grid-cols-1
  lg:grid-cols-3
  md:grid-cols-2
  mt-2
  xl:grid-cols-4
">
  <div>
    <span>
      <a
        aria-label="artist name"
        href="/artists/artist-1/"
      >Artist 1</a>
    </span>
  </div>
</div>
`;
